### PR TITLE
feat: auth0 + profile integration

### LINKED ISSUE #541: Auth state
- [x] Setup Auth0 application
- [x] New redux domain for auth (includes token payload info)
- [x] Dispatch login request action 
- [x] Include login saga
- [x] Support restoring session

### LINKED ISSUE #546: Auth UI
All these tasks can be implemented without a working backend

- [x] Login
- [x] Logout
- [x] Profile (links to Avatars)

### PR DESCRIPTION
**Pull request by cazala**
Closes #541 
Closes #546 

Screen Shot 2019-07-20 at 6 18 12 PM

**To the reviewer** 

There are new `.env` variables:

```
REACT_APP_AVATARS_API_URL=

REACT_APP_AUTH0_CLIENT_ID=
REACT_APP_AUTH0_DOMAIN=
REACT_APP_AUTH0_REDIRECT=
```

### CODE REVIEW

File: src/components/AvatarFace/AvatarFace.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react'
+
+import { bustCache } from './utils'
+import { Props } from './AvatarFace.types'
+// @ts-ignore
+import guestFace from './guest.svg'
```
Comment (belohlavek) [inline src/components/AvatarFace/AvatarFace.tsx]: 🎩svgs via css

File: src/components/ErrorPage/ErrorPage.tsx [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import * as React from 'react'
-import { Button } from 'decentraland-ui'
+import { Button, Page } from 'decentraland-ui'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
 import { getAnalytics } from 'decentraland-dapps/dist/modules/analytics/utils'
 import { IntercomWidget } from 'decentraland-dapps/dist/components/Intercom/IntercomWidget'
 
 import { Props } from './ErrorPage.types'
 import './ErrorPage.css'
+import Navbar from 'components/Navbar'
```
Comment (belohlavek) [inline src/components/ErrorPage/ErrorPage.tsx]: import order 🤓

File: src/components/HomePage/HomePage.tsx [hunk 3]
```diff
@@ -12,6 +12,8 @@ import Icon from 'components/Icon'
 import PromoBanner from './PromoBanner'
 import { Props, State, DefaultProps } from './HomePage.types'
 import './HomePage.css'
+import Footer from 'components/Footer'
```
Comment (belohlavek) [inline src/components/HomePage/HomePage.tsx]: import order 🤓

File: src/components/MobilePage/MobilePage.tsx [hunk 4]
```diff
@@ -9,6 +9,8 @@ import { api, EMAIL_INTEREST } from 'lib/api'
 
 import { Props, State } from './MobilePage.types'
 import './MobilePage.css'
+import Footer from 'components/Footer'
```
Comment (belohlavek) [inline src/components/MobilePage/MobilePage.tsx]: Juanca se quiere matar
Import order 🤓

File: src/components/Navbar/Navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react'
+import { Navbar as DappsNavbar } from 'decentraland-dapps/dist/containers'
+import UserMenu from 'components/UserMenu'
+import { NavbarProps } from 'decentraland-ui'
```
Comment (belohlavek) [inline src/components/Navbar/Navbar.tsx]: Import order 🤓

File: src/components/NotFoundPage/NotFoundPage.tsx [hunk 6]
```diff
@@ -1,11 +1,13 @@
 import * as React from 'react'
-import { Button } from 'decentraland-ui'
+import { Button, Page } from 'decentraland-ui'
 import { getAnalytics } from 'decentraland-dapps/dist/modules/analytics/utils'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
 import { locations } from 'routing/locations'
 
 import { Props } from './NotFoundPage.types'
 import './NotFoundPage.css'
+import Footer from 'components/Footer'
```
Comment (belohlavek) [inline src/components/NotFoundPage/NotFoundPage.tsx]: Import order 🤓

File: src/components/UserMenu/UserMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
+import * as React from 'react'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { Button } from 'decentraland-ui'
+
+import AvatarFace from '../AvatarFace'
+// @ts-ignore
+import addIcon from './add.svg'
+// @ts-ignore
+import switchIcon from './switch.svg'
```
Comment (belohlavek) [inline src/components/UserMenu/UserMenu.tsx]: SVG in css

File: src/components/UserMenu/UserMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+    return (
+      <div className={classes} onBlur={this.handleClose} tabIndex={0}>
+        {isLoggedIn && (
+          <>
+            <div className="toggle" onClick={this.handleToggle}>
+              <AvatarFace size="medium" face={face} />
+            </div>
+            <div className={`menu ${isOpen ? 'open' : ''}`}>
+              <a className="info" href="https://avatars.decentraland.zone">
```
Comment (belohlavek) [inline src/components/UserMenu/UserMenu.tsx]: ENV var?

File: src/components/UserMenu/UserMenu.tsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+export default class UserMenu extends React.Component<Props, State> {
+  state: State = {
+    isOpen: false
+  }
+
+  ref: HTMLElement | null = null
+
+  handleClose = () => {
+    setTimeout(() => this.setState({ isOpen: false }), 100)
```
Comment (belohlavek) [inline src/components/UserMenu/UserMenu.tsx]: Add comment on why are we using the timeout

File: src/components/UserMenu/utils.ts [hunk 10]
```diff
@@ -0,0 +1,3 @@
+export function getFaceUrl(userId: string, cacheBust = false) {
+  return `https://s3.amazonaws.com/avatars-storage.decentraland.org/${escape(userId)}/face.png?cache-bust=${cacheBust ? Date.now() : false}`
```
Comment (belohlavek) [inline src/components/UserMenu/utils.ts]: ENV var?

File: src/modules/auth/sagas.ts [hunk 11]
```diff
@@ -0,0 +1,56 @@
+import { fork, call, all, takeLatest, put, select } from 'redux-saga/effects'
+
+import { AUTH_REQUEST, LOGIN, LOGOUT, authRequest, authSuccess, authFailure } from './actions'
+import { login, logout, handleCallback, restoreSession } from './utils'
+import { isLoggedIn, isExpired } from './selectors'
+import { AuthData } from './types'
+import { api } from 'lib/api'
```
Comment (belohlavek) [inline src/modules/auth/sagas.ts]: import order
¿Cómo se llama el primo vegetariano de Bruce Lee? Pues, Broco Lee.

File: src/modules/translation/languages/en.json [hunk 12]
```diff
@@ -333,5 +333,10 @@
     "promo_title": "Don’t have a wallet yet?",
     "title": "Connect your wallet to decorate your Scene with crypto-collectibles.",
     "signature_error": "Failed to sign message"
+  },
+  "user_menu": {
```
Comment (belohlavek) [inline src/modules/translation/languages/en.json]: missing ZH and ES translations

**Comment by belohlavek** [comment]
Kapture 2019-07-23 at 10 32 33

Flaky avatars link?

**Comment by cazala** [comment]
all the feedback and defects addressed @belohlavek

File: src/modules/auth/sagas.ts (hunk 11 above)
Comment (cazala) [reply src/modules/auth/sagas.ts]: 🥦

**Pull request merged**